src/main/java/com/minecolonies/coremod/util/BuildingUtils.java

```diff
@@ -39,4 +53,48 @@ public static ItemStack getItemStackForHutFromInventory(final Inventory inventor
         }
         return ItemStack.EMPTY;
     }
+
+    /**
+     * Checks a hut block for job= tags to restrict which jobs are allowed to exist there.
+     *
+     * @param world       the world.
+     * @param buildingPos the position of the hut block.
+     * @return            a predicate that returns true if the specified job is permitted.  (note
+     *                    that there may be other reasons the job isn't allowed, such as being the
+     *                    wrong building type for that job; this does not check that.)
+     */
+    @NotNull
+    public static Predicate<JobEntry> getAllowedJobs(@NotNull final Level world, @NotNull final BlockPos buildingPos)
+    {
+        if (world.getBlockEntity(buildingPos) instanceof final IBlueprintDataProviderBE provider)
+        {
+            final Set<String> jobTags = provider.getPositionedTags().getOrDefault(BlockPos.ZERO, new ArrayList<>()).stream()
+                    .filter(t -> t.startsWith("job="))
+                    .map(t -> t.substring(4))
+                    .collect(Collectors.toSet());
+            if (!jobTags.isEmpty())
+            {
+                return job -> jobTags.contains(job.getKey().getPath()) || jobTags.contains(job.getKey().toString());
+            }
+        }
+
+        return job -> true;
+    }
+
+    /**
+     * Check if the given building should try to automatically hire a new citizen.
+     *
+     * @param building   the building to check.
+     * @param hiringMode the current hiring mode of the job.
+     * @param job        the job to hire, or null for a non-specific check.
+     * @return           true if automatic hiring is allowed.
+     */
+    public static boolean canAutoHire(@NotNull final IBuilding building,
+                                      @NotNull final HiringMode hiringMode,
+                                      @Nullable final JobEntry job)
+    {
+        return ((building.getBuildingLevel() > 0 && building.isBuilt()) || building instanceof BuildingBuilder)
```

Comment:
Since you touch this anyway, I kinda feel like, we should actually make this a method in the building, and have an override for the builder.

Reply:
Interestingly, that already exists, but only in the view.  But it probably does make sense to move it to the building instead.